### DISCUSSION TITLE
How to use your own emoji font?

### DISCUSSION THREAD

**Discussion by dignifiedquire** [emilk/egui#1317] [General]
I am trying to use a custom emoji font but just inserting it as a font does not seem to work, any pointers?

**Comment by KentaTheBugMaker** [comment]
Have you tried [this](https://github.com/emilk/egui/blob/master/eframe/examples/custom_font.rs) ?

**Reply by dignifiedquire** [reply]
yes, that’s what I did, but it only shows regular fonts, emojis are not displayed

**Reply by KentaTheBugMaker** [reply]
What is your custom font ?
Do you want to use colored emoji?
epaint store font data in gray scale so we can't use color fonts.

**Reply by dignifiedquire** [reply]
yeah they definitely should be colored 😥

I tried https://openmoji.org/

Are there plans/ideas on adding colored support?

**Reply by KentaTheBugMaker** [reply]
ab_glyph currently support color fonts with PNG data but SVG not supported.

# Additional dependencies to support color fonts.
### PNG bitmap support
* PNG crate
### SVG support
* usvg
* resvg
* tiny_skia 

# epaint extension
* support colored texture atlas(easy)
* some implementation to  `epaint::text::font`
